src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlConnectionStringBuilderTest.cs

```diff
@@ -495,5 +540,56 @@ internal static void CheckEncryptType(SqlConnectionStringBuilder builder, SqlCon
             Assert.IsType<SqlConnectionEncryptOption>(builder.Encrypt);
             Assert.Equal(expectedValue, builder.Encrypt);
         }
+
+        internal static void ExecuteConnectionStringFromJsonThrowsException(string encryptOption)
+        {
+            Assert.Throws<InvalidOperationException>(() => LoadSettingsFromJsonStream<UserDbConnectionStringSettings>(encryptOption));
+        }
+
+        private static TSettings LoadSettingsFromJsonStream<TSettings>(string encryptOption) where TSettings : class
+        {
+            TSettings settingsOut = null;
+
+            Host.CreateDefaultBuilder()
+                .ConfigureAppConfiguration((ctx, configBuilder) =>
+                {
+                    // Note: Inside string interpolation, a { should be {{ and a } should be }}
+                    // First, declare a stringified JSON
+                    var json = $"{{ \"UserDb\": {{ \"UserComponents\": {{ \"NetworkLibrary\": \"DBMSSOCN\", \"UserID\": \"user\", \"Password\": \"password\", \"DataSource\": \"localhost\", \"InitialCatalog\": \"catalog\", \"Encrypt\": \"{encryptOption}\" }}}}}}";
+                   
+                    // Load the stringified JSON as a stream into the configuration builder
+                    configBuilder.AddJsonStream(new MemoryStream(Encoding.ASCII.GetBytes(json)));
+                    configBuilder.AddEnvironmentVariables();
+                })
+                .ConfigureServices((ctx, services) =>
+                {
+                    var configuration = ctx.Configuration;
+                    services.AddOptions();
+                    services.Configure<TSettings>(ctx.Configuration);
+                    settingsOut = configuration.Get<TSettings>();
+                })
+                .Build();
+
+            return settingsOut;
+        }
+    }
+
+    #region ConnectionStringFromJsonRequiredClasses
+    // These 2 classes will be used by ConnectionStringFromJsonTests only
+    internal class UserDbConnectionStringSettings
+    {
+        [Required]
+        public UserSqlConnectionString UserDb { get; set; }
+    }
+
+    internal class UserSqlConnectionString
+    {
+        public SqlConnectionStringBuilder UserComponents { get; set; } = new();
+
+        public override string ToString()
+        {
+            return UserComponents!.ConnectionString;
+        }
     }
+    #endregion
```

Comment:
You can move these lines to the same region with the tests, including the helper function.

Reply:
The helper classes have been moved to the same region with the tests.